flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -1472,15 +1498,35 @@ private ResultPartitionType getResultPartitionType(StreamEdge edge) {
             case HYBRID_SELECTIVE:
                 return ResultPartitionType.HYBRID_SELECTIVE;
             case UNDEFINED:
-                return determineUndefinedResultPartitionType(edge.getPartitioner());
+                return determineUndefinedResultPartitionType(edge);
             default:
                 throw new UnsupportedOperationException(
                         "Data exchange mode " + edge.getExchangeMode() + " is not supported yet.");
         }
     }
 
-    private ResultPartitionType determineUndefinedResultPartitionType(
-            StreamPartitioner<?> partitioner) {
+    private ResultPartitionType determineUndefinedResultPartitionType(StreamEdge edge) {
+        if (outputBlockingNodesID.contains(edge.getSourceId())) {
+            // use corresponding partition type if the upstream node outputs EOF
+            edge.setBufferTimeout(ExecutionOptions.DISABLED_NETWORK_BUFFER_TIMEOUT);
```

Comment:
Suppose operatorA and operatorB is not chained, operatorA sends records to operatorB, and operatorB has outputOnEOF = true, and execution mode is streaming. Will this PR guarantee that the edge between operatorA and operatorB is blocking (or hybrid shuffle), and operatorA runs in batch mode?

If yes, can you provide reference to the code that achieves this? The code in this class seems to only guarantee that those operators chained with an operator with isOutputOnEOF = true will run in batch mode.